deploy/kubernetes/base/controller.yaml

```diff
@@ -16,6 +16,8 @@ spec:
       app.kubernetes.io/name: aws-ebs-csi-driver
   template:
     metadata:
+      annotations:
+        eks.amazonaws.com/compute-type: ec2
```

Comment:
I don't think this is guaranteed to work, need to use a selector or affinity, what if we copied https://github.com/kubernetes-sigs/aws-ebs-csi-driver/pull/488/files?

Reply:
Thank you for your suggestion. I tried it, but it led to `Pending` status of the pods and got a error event:

```
$ kubectl get events -n kube-system

LAST SEEN   TYPE      REASON              OBJECT                                     MESSAGE
<unknown>   Warning   FailedScheduling    pod/ebs-csi-controller-5685df7966-2bsfv    MatchNodeSelector failed: Fargate profile system cannot satisfy pod's node selector/affinity requirements
```

If we create a fargate profile for namespaces, I think any pods in the namespaces should have affinities to nodes which have fargate-node's labels.

Reply:
I think I understand, so kube-system is part of a fargate profile but we still want certain pods in that namespace like this controller or coredns to run on ec2.

Plus, the reason the DaemonSet needs to use affinity is because while fargate on eks already knows to ignore DaemonSets, kube-scheduler doesn't so it will create pointless pods scheduled to fargate nodes.